common/src/main/kotlin/streams/service/Topics.kt

```diff
@@ -26,13 +38,16 @@ data class Topics(val cypherTopics: Map<String, String> = emptyMap(),
             TopicType.PATTERN_NODE to nodePatternTopics, TopicType.PATTERN_RELATIONSHIP to relPatternTopics)
 
     companion object {
-        fun from(config: Map<*, *>, prefix: String, toReplace: String = "", invalidTopics: List<String> = emptyList()): Topics {
-            val cypherTopicPrefix = TopicType.CYPHER.key.replace(prefix, toReplace)
-            val sourceIdKey = TopicType.CDC_SOURCE_ID.key.replace(prefix, toReplace)
-            val schemaKey = TopicType.CDC_SCHEMA.key.replace(prefix, toReplace)
-            val cudKey = TopicType.CUD.key.replace(prefix, toReplace)
-            val nodePatterKey = TopicType.PATTERN_NODE.key.replace(prefix, toReplace)
-            val relPatterKey = TopicType.PATTERN_RELATIONSHIP.key.replace(prefix, toReplace)
+        fun from(map: Map<String, Any?>, replacePrefix: Pair<String, String> = ("" to ""), dbName: String = "", invalidTopics: List<String> = emptyList()): Topics {
+            val config = map
+                    .filterKeys { if (dbName.isNotBlank()) it.endsWith(".to.$dbName") else !it.contains(".to.") }
+                    .mapKeys { if (dbName.isNotBlank()) it.key.replace(".to.$dbName", "") else it.key }
+            val cypherTopicPrefix = if (replacePrefix.first.isNullOrBlank()) TopicType.CYPHER.key else TopicType.CYPHER.key.replace(replacePrefix.first, replacePrefix.second)
```

Comment:
these if/elses are hard to read because of so much repeated code.  Define a helper function that does ```if(replacePrefix.first.isNullOrBlank()) INPUT.key else INPUT.key.replace(replacePrefix.first, replacePrefix.second)```

It'd be very easy for there to be a typo here we would miss

Reply:
Done